docs/reference/modules/page.md

```diff
@@ -0,0 +1,379 @@
+
+### `async archive(req, _id)`
+
+The `archive()` method moves a page, identified by its unique `_id`, into the page tree's archive section. It returns and object with two properties: `parentSlug`, the slug of the page's former parent; and `changed`, an array of objects with `_id` and `slug` properties, identifying all child pages of the moved page that were also archived.
+
+### `async publish(req, draft, options)`
+
+When passed a `req` object and *draft* document object (`draft`), this method will publish the draft. This replaces an existing published version of the page, if there is one. The options object (`options`) is currently only used for internal draft state management.
+
+### `async localize(req, draft, locale, option)`
+
+Localize the draft page (`draft`), copying it to another locale (`locale`). This creates that locale's draft for the first time if necessary. By default existing documents are not updated, but setting `update: true` in the `options` object will update existing ones.
+
+### `async revertDraftToPublished(req, draft)`
+
+Reverts the given draft page (`draft`) to the most recent publication, clearing any changes. It returns the draft's new value, or `false` if the draft was not modified from the published version or no published version exists yet.
+
+Emits the [`afterRevertDraftToPublished` event](/docs/reference/server-events.md#afterrevertdrafttopublished) before returning, which includes the draft document in its payload, allowing you to alter the returned draft object.
+
+### `async revertPublishedToPrevious(req, published)`
+
+Reverts a published page document (`published`) to the previous published state and returns the updated published state. If this was already done (only one previous state is saved) or there is no previous publication, it throws an `invalid` exception.
+
+Emits the [`afterRevertPublishedToPrevious` event](/docs/reference/server-events.md#afterrevertpublishedtoprevious) before returning, which includes the published document in its payload, allowing you to alter the returned published doc object.
+
+### `normalizeSlug(req)`
+
+Normalizes and replaces `req.slug` to account for unneeded trailing whitespace, trailing slashes other than the root, and double slash based open redirect attempts.
+
+### `isPage(doc),`
+
+Returns `true` if the document object, `doc` is identifiable as a page.
+
+
+### `getBaseUrl(req)`
+
+Returns the effective base URL for the given request (`req`). If Apostrophe's top-level `baseUrl` option is set, or a hostname is defined for the active locale, then that is consulted, otherwise the base URL is the empty string. This makes it easier to build absolute URLs (when `baseUrl` is configured), or to harmlessly prepend the empty string (when it is not configured). The Apostrophe queries used to fetch Apostrophe pages consult this method.
```

Comment:
The comment in the code is out of date.

If a hostname is configured for `req.locale`, then the base URL will include it, inferring the protocol from `req.protocol`. Otherwise if `baseUrl` is set it is used. Otherwise the empty string.